### ISSUE TITLE
Raise an error if cannot execute pkgconfig binary used in cross file

### ISSUE THREAD

**Issue by RomanValov** [mesonbuild/meson#7721]
**Describe the bug**
I've run `meson setup` with cross-file, but accidentally has made a typo in `pkgconfig` variable (binaries section).
The `meson` hasn't complained and silently continues to run with native `pkgconfig` resulting into improper configuration.

**To Reproduce**
1) Use cross-file to setup the build
2) Make a typo in `pkgconfig`
3) Run `meson --cross-file ...`

**Expected behavior**
Setup process stops and shows an error about missing pkgconfig

**system parameters**
meson 0.55.1

**Comment by nirbheek** [comment]
>  The `meson` hasn't complained and silently continues to run with native `pkgconfig` resulting into improper configuration.

Yes, the correct approach should be that if the cross `pkgconfig` specified cannot be run, pkg-config cross dependency searching should fail.
